src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripContainerDesigner.cs

```diff
@@ -0,0 +1,484 @@
+                //create our glyph, and set its cursor appropriately
+                ControlBodyGlyph bodyGlyph = new ControlBodyGlyph(translatedBounds, Cursor.Current, currentPanel, panelDesigner);
+                selectionManager.BodyGlyphAdorner.Glyphs.Add(bodyGlyph);
+
+                bool addGlyphs = true;
+                ICollection selComponents = _selectionService!.GetSelectedComponents();
+                if (!_selectionService.GetComponentSelected(_toolStripContainer!))
+                {
+                    foreach (object comp in selComponents)
+                    {
+                        ToolStripContainer container = ContainerParent((Control)comp)!;
+                        addGlyphs = container == _toolStripContainer;
+                    }
+                }
+
+                if (addGlyphs)
+                {
+                    ToolStripPanelDesigner? designer = panelDesigner as ToolStripPanelDesigner;
+                    if (designer is not null)
+                    {
+                        AddPanelSelectionGlyph(designer, selectionManager);
+                    }
+                }
+            }
+        }
+
+        return base.GetControlGlyph(selectionType);
+    }
+
+    private static Control? GetAssociatedControl(Component component)
+    {
+        Control? associatedControl = null;
+        if (component is Control control)
+        {
+            return control;
+        }
+
+        if (component is ToolStripItem item)
+        {
+            Control parent = item.GetCurrentParent()!;
+            parent ??= item.Owner!;
+
+            return parent;
+        }
+
+        return associatedControl;
+    }
+
+    private bool CheckDropDownBounds(ToolStripDropDownItem dropDownItem, Glyph childGlyph, GlyphCollection glyphs)
+    {
+        if (dropDownItem is null)
+        {
+            return false;
+        }
+
+        Rectangle glyphBounds = childGlyph.Bounds;
+        Rectangle controlBounds = BehaviorService.ControlRectInAdornerWindow(dropDownItem.DropDown);
+        if (!glyphBounds.IntersectsWith(controlBounds))
+        {
+            glyphs.Insert(0, childGlyph);
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    ///  Checks if the associated control bounds overlap the PanelSelectionGlyph bounds.
+    /// </summary>
+    private bool CheckAssociatedControl(Component component, Glyph childGlyph, GlyphCollection glyphs)
+    {
+        bool result = false;
+
+        ToolStripDropDownItem? item = component as ToolStripDropDownItem;
+        if (item is not null)
+        {
+            result = CheckDropDownBounds(item, childGlyph, glyphs);
+        }
+
+        if (!result)
+        {
+            Control? associatedControl = GetAssociatedControl(component);
+            if (associatedControl is not null && associatedControl != _toolStripContainer)
+            {
+                if (!PInvoke.IsChild((HWND)_toolStripContainer!.Handle, (HWND)associatedControl.Handle))
```

Comment:
I believe you can simplify to the following
```suggestion
                if (!PInvoke.IsChild(_toolStripContainer!, associatedControl)
```